engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -131,9 +133,13 @@ public WorldRendererLwjgl(WorldProvider worldProvider, ChunkProvider chunkProvid
         }
         localPlayerSystem.setPlayerCamera(playerCamera);
 
+        initMainDirectionalLight();
+
         renderableWorld = new RenderableWorldImpl(worldProvider, chunkProvider, bufferPool, playerCamera, shadowMapCamera);
         renderQueues = renderableWorld.getRenderQueues();
+    }
 
+    private void initMainDirectionalLight() {
         mainDirectionalLight.lightType = LightComponent.LightType.DIRECTIONAL;
```

Comment:
I agree, and I will certainly aim to have light components on celestial body entities. Ok, will leave as it is for the time being.